### ISSUE TITLE
questions of inputing my own point cloud

### ISSUE THREAD

**Issue by FengenYang** [guohaoxiang/ComplexGen#10]
Hi, haoxiang,

thanks for sharing such a great work, now I'm trying to follow what you mentioned in #3 to input my own point cloud, and I met some problems.

I got the same error after dumping the data_list into the pkl file as Eusford did in #7 :
![image](https://user-images.githubusercontent.com/23524235/223365437-fac7abf4-fbbb-4894-9c2a-e2d19d333ba3.png)

As you said, data in my own pkl file should be a list not a dictionary, however, when data_loader_abc.py is loading my pkl file, the script tries to get the content of "filename" as the error shows. If it's a list, the index should be a number rather than the string "filename", I think this is why Eusford and I met the same error. So I wonder am I taking it in a wrong way?

Then I tried your train_small/*.pkl file and only change the "surface_points" to my own point cloud, as you said only this part is used for inference, but I visualized the predicition output, I find the "curves" and "patches" are still affecting my point cloud, the output is shown below:
![image](https://user-images.githubusercontent.com/23524235/223369198-c89edfe0-042d-4534-a854-9f534340cd6e.png)
So I want to ask that is it correct to only replace "surface_points" and keep the old "curves" and "patches"?

I'll be appreciate if you can help me with these two questions!
Best regards,
Enyang Feng

**Comment by guohaoxiang** [comment]
Can you tell me the exact commands you used for inference after replacing "surface_points" ?

**Comment by guohaoxiang** [comment]
By the way, can you pack the original data/train_small folder and send it to me?

**Comment by guohaoxiang** [comment]
Besides, please provide your prepared data to me. Thanks!

**Comment by FengenYang** [comment]
Sure, here are my related files and please allow me to show you my whole process which may make my problem clear:

1. At first, I normalized my pointcloud "torus1.pcd" in [-0.5,0.5]^3. By the way, Im not sure if I'm doing this correctly because this will change the shape of my pointcloud:
```
  import open3d as o3d
  import numpy as np
  
  pcd=o3d.io.read_point_cloud('torus1.pcd')
  
  #normalize
  xmax=(pcd.points)[np.argmax((pcd.points),axis=0)[0]][0]
  xmin=(pcd.points)[np.argmin((pcd.points),axis=0)[0]][0]
  ymax=(pcd.points)[np.argmax((pcd.points),axis=0)[1]][1]
  ymin=(pcd.points)[np.argmin((pcd.points),axis=0)[1]][1]
  zmax=(pcd.points)[np.argmax((pcd.points),axis=0)[2]][2]
  zmin=(pcd.points)[np.argmin((pcd.points),axis=0)[2]][2]
  
  for i in (pcd.points):
      i[0]=(i[0]-xmin)/(xmax-xmin)-0.5
      i[1]=(i[1]-ymin)/(ymax-ymin)-0.5
      i[2]=(i[2]-zmin)/(zmax-zmin)-0.5
  
  points=np.asarray(pcd.points)
  normals=np.asarray(pcd.normals)
  input_point=np.concatenate((points,normals),axis=1)
```

2. I load the pkl file "packed_000000.pkl"in "train_small" folder, change the surface_points in the first dictionary and save this dictionary in my own pkl file "input.pkl".

```
import pickle
read_pkl=open('packed_000000.pkl','rb')
write_pkl=open('input.pkl','wb')
dump_data=pickle.load(read_pkl)

dump_data['surface_points']=input_point
pickle.dump(dump_data,write_pkl)

read_pkl.close()
write_pkl.close()
```

3.  I put my own pkl file "input.pkl" into ComplexGen/data/default/test/packed/ . Then following your instructions, I run command `./scripts/test_default.sh` and './scripts/extraction_default.sh' to get extraction.complex file. In Windows system, I run `./scripts/geortric_refine.py` to get json file. At last, I use gen_vis_result.py to get the obj file.

Thanks for your help!
[files.zip](https://github.com/guohaoxiang/ComplexGen/files/10916902/files.zip)

**Comment by guohaoxiang** [comment]
HI,  I've update the input data preparation procedure in https://github.com/guohaoxiang/ComplexGen/issues/3. I think you are doing the right thing except that the normals of your input point cloud are all zero. Please prepare a point cloud with correct normals for testing.

**Comment by FengenYang** [comment]
Thanks for your help! I'll try to change the normals for testing. And I have another question, recently I'm using your pre-processed ABC dataset for testing, I followed your instructions : test_default.py -> extraction_default.py -> geometric_refine.py. At last, I visualized the generated json file which are shown as follow, these B-Rep models looks a little strange， are they normal? I  found you said you didn't release the trimming step code in #9, does it cause my problem?

![image](https://user-images.githubusercontent.com/23524235/224592985-eee0696c-6429-4566-b2ee-fb7234b45f11.png)
00866067_0_extraction_input

![image](https://user-images.githubusercontent.com/23524235/224593055-625d2128-6d37-44aa-8367-28ea45fe7d04.png)
00866904_0_extraction_input

![image](https://user-images.githubusercontent.com/23524235/224593138-0e7e8432-016d-48c9-ba18-34cc1ccddc23.png)
00869648_2_extraction_input
